### PR TITLE
Bug 1955221: implement a switch in ListV1 and ListV2

### PR DESCRIPTION
**Pull request by alaypatel07**

### CODE REVIEW

**Review by sseago — APPROVED** [review]

File: velero-plugin-for-aws/object_store.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ import (
 	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/aws/aws-sdk-go/service/s3/s3manager"
 	"github.com/pkg/errors"
+	"github.com/prometheus/common/log"
```
Comment (djwhatle) [inline velero-plugin-for-aws/object_store.go]: I think a new logger (prometheus common log) is being introduced here. Velero appears to use logrus. Intentional?

File: velero-plugin-for-aws/object_store.go [hunk 2]
```diff
@@ -350,17 +352,45 @@ func (o *ObjectStore) ListCommonPrefixes(bucket, prefix, delimiter string) ([]st
 		Delimiter: &delimiter,
 	}
 
+	tryListV1 := false
 	var ret []string
 	err := o.s3.ListObjectsV2Pages(req, func(page *s3.ListObjectsV2Output, lastPage bool) bool {
 		for _, prefix := range page.CommonPrefixes {
 			ret = append(ret, *prefix.Prefix)
 		}
+		if *page.IsTruncated && page.NextContinuationToken == nil {
```
Comment (djwhatle) [inline velero-plugin-for-aws/object_store.go:368]: I'm not knowledgeable on the behavior of this API, but could you have a page that is truncated due to being the final page of results that wouldn't have a continuation token when everything is working as expected?

Or is a page only considered "truncated" when there are supposed to be more results on continued pages?

**Review by djwhatle — COMMENTED** [review]
Left some comments. I haven't been close to this issue so better that someone else reviews. Looks like @sseago already gave his ACK 👍

File: velero-plugin-for-aws/object_store.go (hunk 1 above)
Comment (alaypatel07) [reply velero-plugin-for-aws/object_store.go]: not intentional, IDE screwed me :D I will post an update to this.

File: velero-plugin-for-aws/object_store.go (hunk 2 above)
Comment (alaypatel07) [reply velero-plugin-for-aws/object_store.go:368]: @djwhatle truncated is true when there can be more results but the server couldn't send it on this page.

File: velero-plugin-for-aws/object_store.go (hunk 1 above)
Comment (alaypatel07) [reply velero-plugin-for-aws/object_store.go]: updated, PTAL

**Review by sseago — APPROVED** [review]

**Pull request merged**